proposals/pr002-hunkspace.md

```diff
@@ -0,0 +1,545 @@
+# CLDR "Hunkspace"
+
+2020-07-09
+
+Author: [filmil (Filip Filmar) · GitHub][ff]
+
+*tl;dr:* Generalize the [ICU4X Data Provider][dp] key space to include
+localized data from sources other than [CLDR][cldr] More detail is in the
+[Detailed Proposal](#h.us1rqruwdp1h) section, after the proposal is motivated
+by a few use cases.
+
+This document proposes a global, shared *hunkspace* (better names always
+welcome) of localized "[hunk][hunk]" data for the ICU4X Data Provider.  It
+doesn't necessarily affect the ICU4X data provider implementation, but does
+affect the data model.   Specifically the proposal is to make the data model
+more general and open to independent third party extensions.
+
+> The terminology proposed in [ICU4X Locale Data Pipeline][icudp] is used.  Any
+> inconsistent word use is a bug, and should be fixed.
+
+[cldr]: http://cldr.unicode.org
+[dp]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+[ff]: https://github.com/filmil
+[hunk]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+[icudp]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+
+# Prior Art
+
+The *ICU4X Locale Data Pipeline* document proposes a [data model][dm] that fits
+the current use case of CLDR data. The data storage method is abstracted from
+the API by using *Data Providers*.  The API is using the
+`key:request/value:response` pattern.  See [Request][rq] in the source code for
+the example key.
+
+[dm]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+[rq]: https://github.com/unicode-org/icu4x/blob/master/components/data-provider/src/lib.rs
+
+# Use cases
+
+> This section lists a few use cases where a shared hunkspace are helpful.
+
+## Third party language packs
+
+The use case comes from [Fuchsia OS][fx], though it is not a
+unique need. The explanation below gives the context of the use case, and ends
+with the description of the case itself.
+
+Fuchsia software is *ephemeral* (downloaded on demand) and is
+*trusted* (verified prior to execution). So, if a process runs on Fuchsia,
```

Comment:
nit: formatting: don't put italics directly into the md file if you're going to use the `* *` syntax (probably some renderer error, unsure of how you could have come across this yourself)

Reply:
I think the renderer stylizes the text between asterisks.  This is not really there in the text, nor do I know how it could be there.